src/main/java/org/datadog/jmxfetch/Instance.java

```diff
@@ -228,6 +236,9 @@ public String toString() {
                 }
             }
         }
+        if (this.minReturnedMetrics < metrics.size()) {
```

Comment:
If we decide to make `minReturnedMetrics` `final` then we will have to initialize it to an integer (we could set a `-1` to reflect the option is disabled) in the constructor. If we don't make it `final` then we could check it's null before making this comparison or else I think we could run into an exception here.